File: Homework/Homework_3/zhou-ke-xuan/learn/test2.py
import cv2
import numpy as np

# 读取图像（OpenCV默认读取为BGR格式）
img_bgr = cv2.imread("image.png")

# # 转换为RGB格式
img_rgb = cv2.cvtColor(img_bgr, cv2.COLOR_BGR2RGB)

# 分离RGB通道
r, g, b = cv2.split(img_rgb)

# 合并RGB通道
img_red = cv2.merge([r, g, b])

# 显示图像
cv2.imshow("Original Image", img_bgr)
cv2.imshow("Red Image", img_red)
cv2.waitKey(0)
cv2.destroyAllWindows()

File: Homework/Homework_3/zhang-guo-yuan/test 2/test2.cpp
#include <opencv2/core.hpp>
#include <opencv2/highgui.hpp>
#include<opencv2/imgproc.hpp>
#include<iostream>
int main(){
    cv::Mat img_bgr = cv::imread("../image.png");
    
    if(img_bgr.empty()){
        std::cerr << "Error : Unable to load image!"<<std::endl;
        return -1;
    }
    cv::Mat img_rgb;
    cv::cvtColor(img_bgr , img_rgb,cv::COLOR_BGRA2BGR);
    
    std::vector<cv::Mat>channels(3);
    cv::split(img_rgb , channels);

    cv::Mat img_red;
    cv::merge(channels , img_red);

    cv::imshow("Originak Image" , img_bgr);
    cv::imshow("Red Image" , img_red);

    cv::waitKey(0);
    cv::destroyAllWindows();

    return 0;
}